### ISSUE TITLE
File drag and drop support for Linux

### ISSUE THREAD

**Issue by nbolton** [deskflow/deskflow#4170]
Problem: We never implemented Linux support for file drag and drop, which kind of sucks.

Solution: Make those Linux users feel loved!

https://github.com/symless/synergy/issues/155

**Comment by iwarford** [comment]
Agreed.  I was surprised when it didn't work.  This is essential.

**Comment by q13** [comment]
hope so!

**Comment by ucpete** [comment]
This is the *only* feature Synergy doesn't have that I truly want. Everything else is just perfect. Thanks!

**Comment by mixu-** [comment]
I really want this feature as well.

**Comment by danShumway** [comment]
+1 Synergy is already kind of black magic to me, but this feature would be amazing.

**Comment by nicolasticot** [comment]
I agree. Such capability is needed. I am surprise that this is more difficult for Linux than for Mac. This subject should take a high priority. In a sense I find very important to get all the same capability for different Windows/MAC/Linux. Actually I have to cross Linux to go from one Windows machine to the other. Drag and drop it thus not working.

**Comment by claell** [comment]
Any updates?

**Comment by rknall** [comment]
+1 from me, desperately needed.

**Comment by afranke** [comment]
Please don't leave "+1" or "ping" comments as they only add noise to bug reports. If you want to show interest in a feature request, use the :+1: at the bottom of issue description.

**Comment by rknall** [comment]
@afranke - done

**Comment by Trollwut** [comment]
Do we have any update on this feature request?

E.g. that Linux-to-Linux i in testing and Linux-to-* has to be programmed.

This feature would substitute my solution of using a network drive as a temporal storage, which is rather unconvenient if moving different files of different directories.

**Comment by nbolton** [comment]
Hey everyone, we really want to get this implemented. We have a huge workload on and not enough people, but we're doing our best. If you're looking for a job: symless.com/careers

**Comment by c-hartmann** [comment]
Drag'n Drop and/or Copy'n Paste from (Any incl. Linux) Desktop to (Any incl. Linux) Desktop would be nice as a start of implementation. Hooking into every f* API in the Linux world might be pain in the ass .. i agree. So Desktop to Desktop is the thing to target?

**Comment by coxbrandon** [comment]
People have been begging for this feature for _literally_ years.  It would be nice if this was finally prioritized accordingly.

**Comment by afranke** [comment]
> It would be nice if this was finally prioritized accordingly.

It would also be nice if people stopped adding comments just to complain and understood that

* developing features takes time
* there are many features to implement and bugs to fix, so developers have to juggle with priorities
* showing interest in a particular issue being resolved can be best achieved by using a thumbs up on the description, not by adding a comment that sounds like a rant and will only demotivate the developers

As a side note, the fact that @nbolton created this issue is proof on its own that @symless does care about this being resolved, so really there is no need poking them again and again about it.

**Comment by lig** [comment]
@afranke a lot of people out there feel kinda deceived. Years ago this was announced as a premium feature which actually works. And there is no mention still except these bugs that this feature doesn't work on Linux. One saw platform support: "Windows/Mac/Linux" and drag-n-drop as a premium feature and paid exactly because of this one. Back in the days when there were donations asked by Synergy, I donated exclusively because of this feature.
And what we keep seeing here? Years of moving the planned date for this.
This is just ridiculous.

**Comment by e-alfred** [comment]
@lig You are very right, it is just that.

The website is still deceiving by saying that "Drag and drop files" works on all platforms without 
mentioning that in reality it only works on Windows/OS X. I mentioned this a long time ago on another issue and nothing was changed, the shady practices to gain more unsuspecting users by not mentioning this are still there. 

Is it really that difficult to change the website by adding an annotation (a * and a description below would suffice) that this feature is not supported on Linux?

**Comment by nickinvenco** [comment]
PLEASE GIVE US THIS FEATURE !

**Comment by sciencedude100** [comment]
totally need this (Ubuntu Budgie 17.04)

**Comment by SwooshyCueb** [comment]
I can't tell you how many times I've watched a YouTube video that had a segment that started out with something to the effect of "This video was brought to you by Synergy." These segments are deceptive, as they deceive users into believing that this feature exists across all platforms. This is a really shitty practice.

**Comment by murlou** [comment]
I was one of the unsuspecting customers
https://symless.com/synergy/pricing
Pricing website still states that drag and drop works without restrictions...
![image](https://user-images.githubusercontent.com/1576368/29485812-9607b6b6-8513-11e7-9658-fb0eb90187c7.png)

**Comment by Doonut** [comment]
Wow its been 3 years and you cant have a feature you advertise? comon...

**Comment by SwooshyCueb** [comment]
I recently got an email marketing an upgrade to Synergy 2. It kinda rubs me the wrong way that you're trying to sell me an upgrade when I still haven't gotten everything I already paid for.

**Comment by iwarford** [comment]
Any progress on this at all?  It's been *years*.

**Comment by sciencedude100** [comment]
anything??? This is seriously an issue

**Comment by CelestiorTheGreat** [comment]
Seriously?? I was willing to buy Synergy 2 to make it easier to work between my linux workstation and my windows based laptop. But this ruins it... Not to mention that it's clearly false advertising.

Welp, I guess I either crack synergy 2 or stick with the free 1.8.8 :)

**Comment by lig** [comment]
Looks like this is the time when Synergy should either do something with Linux support or drop it at all officially.

**Comment by Trollwut** [comment]
Or at least look at this thread once :>

**Comment by ruba** [comment]
well, this is kinda disappointing - just bought it since I need some solution for work between linux and windows 👎

**Comment by astrotk** [comment]
The Synergy website still touts the drag and drop. I wish I would have found this post before I bought it because I need it to work with Windows and Mint.  I agree, fix it or change what you advertise.

**Comment by LeadManPL** [comment]
@nbolton 
Hey Nick,
the [link](https://github.com/symless/synergy/issues/155)  you provided in this issue leads to 404 error page. 
The same case is for issue #4090

**Comment by MortimerNeuss** [comment]
This Thread is 4.5Y old. And the Feature is still not in the Product. I will not buy the product or your excuse. This is just stupid.

**Comment by coxbrandon** [comment]
> This Thread is 4.5Y old. And the Feature is still not in the Product. I will not buy the product or your excuse. This is just stupid.

Do we all wish this had been fix and/or marketed correctly?  Absolutely.  But this is such a juvenile response and does little to further the conversation.  It is still a fantastic, feature-rich product that I will continue to support (I have purchased licenses for both 1.X and 2.X).  Just create a samba share on your Linux desktop like the rest of us until this gets addressed (if it ever does).

**Comment by C-Kenny** [comment]
Can't wait for this to be implemented.

**Comment by cnlohr** [comment]
I think there may even be a middleground if there's something fundamental thing making this issue particularly difficult, i.e. selecting a file on the Linux from the synergy GUI window side pastes to some folder on the Windows side or something to that effect.

**Comment by wrapper** [comment]
Hello from 2021 :)

**Comment by luochen1990** [comment]
Hello from 2022,   I donated for synergy1 and paied for synergy2 years ago,  but still don't have this feature  :(

**Comment by yakobie** [comment]
please make this a thing, having a steam deck with drag and drop file transfers would be amazing

**Comment by dogmatic69** [comment]
9 years later, still not feeling the love. 💔

**Comment by nbolton** [comment]
We do plan to implement this. Sorry that this issue has been open for so long. We are a small team with limited resources.

**Comment by sithlord48** [comment]
Imho this is out of the project scope , we have a broken impl of drag and drop and there are better ways to do filesharing then this.

**Comment by Doonut** [comment]
No, this was an advertised feature on the front page of synergy.com for years. It doesn't matter that is is out of the "project scope". Users are still waiting for a feature they paid for and clearly want.

**Comment by sithlord48** [comment]
deskflow != synergy

**Comment by SwooshyCueb** [comment]
That's an interesting way to dodge accountability.

**Comment by sithlord48** [comment]
Accountability is mine only since deskflow has been separated from synergy. I have nothing todo with anything on their page or what they sold or did not sell you in the past.

**Comment by AwwwwHamburgers** [comment]
Maybe explaining what has transpired would be a better approach than acting crass and expecting everybody who has been following this issue and waiting for a decade to expect to know what has recently happened.

**Comment by sithlord48** [comment]
Deskflow is not synergy , its a new project born from the ashes other the old synergy core mixed with my hopes for things i wanted to do in input-leap. Since trying to do anything in input-leap was met with constant road bocks. I was about to fork it and instead ended up taking over this project. 

While synergy will still be downsteam of deskflow, deskflow is not bound by any promises made by synergy its a completely open source project driven by the members of the community not what someone thinks is best for sales (or whatever companies do).  I have been working to improve the old code base and clean out all the rot to make a more stable base going forward. And as said above the feature is broken on every os (if it even did work on any of them to begin with) and i think its outside the scope of what deskflow should be and that is the core features that is stable keyboard mouse sharing with clipboard support.  Anything outside of that can be done already with better tools. synergy may have (or add) this feature. but for now with the limited resources we have in this project (developer wise ) Id rather use that to fix our core and get it into a more maintainable state then to chase features we don't have time to full bake.

I understand the frustration around the feature even if i don't understand why ppl want this myself. It sucks to be told its coming and it never does.. I'm sorry that what happened but I am not the one responsible for it.

**Comment by AwwwwHamburgers** [comment]
Thank you.  The project changing hands was completely transparent to us so I (and I'm sure most the rest of us) were still under the impression this was still synergy, since we've been following this particular issue for many, many years.

So with the above in mind, I have little issue with this being deemed out of scope, despite many years of hope :-(

**Comment by sithlord48** [comment]
No I totally get. I'm just often to busy coding to write those documents.

**Comment by nbolton** [comment]
FWIW, we do have file-sharing features planned in Synergy 3, including drag and drop. In Synergy 1 (based on Deskflow), the old macOS/Windows drag and drop code is still there but we plan to replace it with something better, and something that works on Linux of course (which is my primary dev OS). Whatever approach we take will most likely use open source technology under the hood, but will likely be a separate project to the Synergy core code.

As @sithlord48 has mentioned though, we have to be very careful with the scope of Deskflow and so anything beyond keyboard, mouse, and clipboard sharing will be out of scope.

@sithlord48 If you are certain about dropping drag and drop entirely and don't mind making people on this thread unhappy, you may want to consider closing this issue as "won't fix".

**Comment by sithlord48** [comment]
I've intentionally not marked it won't fix for the time being

**Comment by sithlord48** [comment]
Related #5419
